### PR TITLE
#112 - reconcile removed associations on subsequent upserts

### PR DESCRIPTION
**Pull request by tshedor**

### CODE REVIEW

File: packages/brick_sqlite_generators/lib/src/sqlite_serialize.dart [hunk 1]
```diff
@@ -204,6 +210,15 @@ class SqliteSerialize<_Model extends SqliteModel> extends SqliteSerdesGenerator<
 
     return '''
       if (instance.${InsertTable.PRIMARY_KEY_FIELD} != null) {
+        final ${field.name}OldColumns = await provider?.rawQuery('SELECT `$joinsForeignColumn` FROM `$joinsTable` WHERE `$joinsLocalColumn` = ?', [instance.${InsertTable.PRIMARY_KEY_FIELD}]);
+        final ${field.name}OldIds = ${field.name}OldColumns?.map((a) => a[$joinsForeignColumn]) ?? [];
```
Comment (polRk) [inline packages/brick_sqlite_generators/lib/src/sqlite_serialize.dart:214]: `a['$joinsForeignColumn']`

**Review by polRk — CHANGES_REQUESTED** [review]
Missing single quote

File: packages/brick_sqlite_generators/lib/src/sqlite_serialize.dart (hunk 1 above)
Comment (tshedor) [reply packages/brick_sqlite_generators/lib/src/sqlite_serialize.dart]: @polRk added

**Review by polRk — APPROVED** [review]
In my project, it works!

**Comment by tshedor** [comment]
@polRk I updated this just a little bit - since calling `.clear()` on an instance is permitted in Dart but isn't true to the ideal of treating `final` as immutable, for associations where you want to call `.clear()` you should be sure to specify those fields as `var fieldName` or `List<YourAssociationName> fieldName`.

**Pull request merged**